client/js/widgets/label.js

```diff
@@ -46,14 +48,23 @@ export class Label extends Widget {
       this.input.style.overflowY = contentHeight-this.get('height') < 5 ? 'hidden' : 'scroll';
     }
 
+    if(delta.placeholderText !== undefined) {
+      this.input.setAttribute('placeholder', this.get('placeholderText'));
+    }
     if(delta.editable !== undefined) {
-      if(delta.editable)
+      if(delta.editable) {
         this.input.removeAttribute("readonly");
-      else
+        if(delta.tabIndex !== undefined) {
+          this.input.setAttribute('tabindex', this.get('tabIndex'));
+        }
+      } else {
         this.input.setAttribute("readonly", true);
+        this.input.removeAttribute("tabindex");
+      }
     }
```

Comment:
You only update `tabindex` if the delta _also_ contains `editable`.

Reply:
Isn't that what this does already?
if delta.editable:
   Remove readonly.
   then, If tabindex is also set add it.
   
Otherwise:
   Make it readonly
   and remove the tabindex.

Reply:
It only adds the tab index if delta.editable.

Reply:
I'm saying that your code doesn't change anything if the widget is already `editable` and then you set `tabIndex`. Your code will only execute if you change `editable`; not if you only change `tabIndex`.

Reply:
Gotcha. Submitted a fix. Now if tabindex is set it adds it, then relies on the editable check to remove it if it's not editable, otherwise it leaves it alone.

Reply:
Is it a problem to have a `tabindex` when it is `readonly`? I would guess that the browser only focuses editable textareas anyway.

Reply:
> Is it a problem to have a `tabindex` when it is `readonly`? I would guess that the browser only focuses editable textareas anyway.

My browser puts the focus on every text area and maybe some other stuff.  So every label in the room gets the focus as I press TAB.  The advantage here I guess is that once on a widget with `tabIndex` it goes through them in order.

Tested the new changes and everything still works.

Reply:
I would say that you really only want keyboard tab navigation to elements that are not readonly as focusing on them isn't terribly worthwhile since you cannot interact with them so it would only serve to add extra tab steps getting to the stuff you can actually interact with via the keyboard.

Reply:
Which may be a good argument for the default being `tabindex="-1"` for all readonly elements.

Reply:
I suppose we can try it and if there's any feedback on it we can always re-visit it. I don't think this behavior will be game-breaking in any way and reversing it shouldn't break any future games that use `tabIndex`.

Reply:
I wanna tweak a thing or two tomorrow.